shared/actions/authActions.js

```diff
@@ -153,71 +188,182 @@ export const setToken = (token) => {
   }
 };
 
-export const getUserToken = async () => {
-  try {
-    return await AsyncStorage.getItem(STORAGE_KEY_accessToken);
-  } catch (error) {
-    console.log(error.message);
+export const setUserName = (userName) => {
+  return {
+    type: types.setUserName,
+    userName
   }
 };
 
-export const getSecretToken = async () => {
-  try {
-    return await AsyncStorage.getItem(STORAGE_KEY_secret);
-  } catch (error) {
-    console.log(error.message);
+export const setUserEmail = (userEmail) => {
+  return {
+    type: types.setUserEmail,
+    userEmail
   }
 };
 
-export const getRefreshToken = async () => {
-  try {
-    return await AsyncStorage.getItem(STORAGE_KEY_refreshToken);
-  } catch (error) {
-    console.log(error.message);
+export const setProfileImgUrl = (profileImgUrl) => {
+  return {
+    type: types.setProfileImgUrl,
+    profileImgUrl
   }
 };
+// todo : pass it to grantfbuser after receiving 400
+export const requestRefreshTokenFacebook = async (refreshToken) => {
+  const address = 'https://goober.herokuapp.com/api/auth/refresh';
+  const bodyRequestRefreshToken = JSON.stringify({
+    'refreshToken': refreshToken
+  });
+  fetch(address, {
+    method: 'POST',
+    headers: {
+      'Accept': 'application/json',
+      'Content-Type': 'application/json'
+    },
+    body: {
+      'refreshToken': bodyRequestRefreshToken
+    }
+  })
+  .then((response) => {
+    if (response.status === 200) {
+      return response.json();
+    } else if (response.status === 400) {
+      removeUserToken();
+    }
+  })
+  .then((rjson) => {
+    console.log(rjson);
+    setAccessToken(rjson.accessToken);
+    setRefreshToken(rjson.refreshToken);
+  })
+  .catch((error) => {
+    console.log(error);
+  })
+};
 
-export const setUserToken = async (response, userType) => {
-  try {
-    await AsyncStorage.setItem(STORAGE_KEY_accessToken, response.accessToken);
-    await AsyncStorage.setItem(STORAGE_KEY_refreshToken, response.refreshToken);
-    if (userType === 'anonymous') {
-      await AsyncStorage.setItem(STORAGE_KEY_secret, response.secret);
+export const requestRefreshTokenGuest = async (refreshToken) => {
+  const address = 'https://goober.herokuapp.com/api/auth/refresh';
+  const bodyRequestRefreshToken = JSON.stringify({
+    'refreshToken': refreshToken
+  });
+  fetch(address, {
+    method: 'POST',
+    headers: {
+      'Accept': 'application/json',
+      'Content-Type': 'application/json'
+    },
+    body: bodyRequestRefreshToken
+  })
+  .then((response) => {
+    if (response.status === 200) {
+      return response.json();
+    } else if (response.status === 400) {
+      getSecretToken().then((secret) => {
```

Comment:
I  think you can bind this two conditional statement with one promise function,
because userKey and userSecret are both required to grant anonymous user.
so if you make this as a function ( that checkout userKey and secret token is both exist and if they does returns them), this flow would be much more readable.